package/MDAnalysis/analysis/base.py

```diff
@@ -447,8 +479,39 @@ def run(self, start=None, stop=None, step=None, frames=None,
             self.times[i] = ts.time
             self._single_frame()
         logger.info("Finishing up")
+        return self
+
+    def _setup_bslices(self, start=None, stop=None, step=None, frames=None):
+        self._bslices = [(start, stop, step, frames)]
+    
+    def _setup_scheduler(self, **kwa):
+        self._scheduler_kwargs = kwa
+
+    def run(self, start=None, stop=None, step=None, frames=None,
+            verbose=None, *, progressbar_kwargs={},
+            scheduler=None):
+        if scheduler is None: # fallback to the local scheduler
+            self._compute(start=start, stop=stop, step=step, frames=frames, 
+            verbose=verbose, progressbar_kwargs=progressbar_kwargs)
+        else: # elif scheduler is a type of dask scheduler
+            self._setup_scheduler()
+            self._setup_bslices(start=start, stop=stop, step=step,
+                frames=frames)
+            computations = delayed(
+                [
+                    delayed(self.run)(start=bstart, stop=bstop, step=bstep, frames=bframes)
```

Comment:
maybe explicitly write `scheduler=None`?

Reply:
good idea!

Reply:
done

Reply:
actually, now I call `_compute` instead of `run` (which makes much more sense), and `scheduler` keyword is not used there.